src/ProgramBuilder.ts

```diff
@@ -68,7 +68,10 @@ export class ProgramBuilder {
     private staticDiagnostics = [] as BsDiagnostic[];
 
     public addDiagnostic(srcPath: string, diagnostic: Partial<BsDiagnostic>) {
-        let file: BscFile = this.program.getFile(srcPath);
+        if (!this.program) {
+            throw new Error('Cannot call `ProgramBuilder.addDiagnostic` before `ProgramBuilder.run()`');
+        }
```

Comment:
An error would be thrown anyway in this case, but I a) wanted to satisfy TS and b) hit this error and was confused by it while writing a plugin in the past, so I thought an explicit error message might be nice.

Reply:
yeah this is fine.